Classes/Console/Service/Database/SchemaService.php

```diff
@@ -43,10 +43,23 @@ public function __construct(SchemaUpdateInterface $schemaUpdate)
      */
     public function updateSchema(array $schemaUpdateTypes, $dryRun = false)
     {
-        $updateStatements = [
-            SchemaUpdateType::GROUP_SAFE => $this->schemaUpdate->getSafeUpdates(),
-            SchemaUpdateType::GROUP_DESTRUCTIVE => $this->schemaUpdate->getDestructiveUpdates(),
+        $select = [
+            SchemaUpdateType::GROUP_SAFE => false,
+            SchemaUpdateType::GROUP_DESTRUCTIVE => false,
         ];
+        foreach ($schemaUpdateTypes as $schemaUpdateType) {
+            foreach ($schemaUpdateType->getStatementTypes() as $statementGroup) {
+                $select[$statementGroup] = true;
+            }
+        }
+
+        $updateStatements = [];
+        if ($select[SchemaUpdateType::GROUP_SAFE]) {
+            $updateStatements[SchemaUpdateType::GROUP_SAFE] = $this->schemaUpdate->getSafeUpdates();
+        }
+        if ($select[SchemaUpdateType::GROUP_DESTRUCTIVE]) {
+            $updateStatements[SchemaUpdateType::GROUP_DESTRUCTIVE] = $this->schemaUpdate->getDestructiveUpdates();
+        }
 
```

Comment:
IMO this could be shortened by directly populating `$updateStatements` depending on the `$schemaUpdateTypes`, maybe using `match()`. No need for boolean flags then.

Reply:
I know... this is the state of the issue. 
I will change this shortly....
Thanks for your fast feedback.

Reply:
@lochmueller Thanks a bunch for the PR! It looks good to me honestly.

@mbrodala I think the whole code could benefit from a refactor, but I would be fine to fix the issue first and refactor everything later in an additional PR

Reply:
Agree with @helhum, refactorings can follow incrementally.